app/src/main/java/space/taran/arknavigator/ui/fragments/utils/Preview.kt

```diff
@@ -1,24 +1,28 @@
 package space.taran.arknavigator.ui.fragments.utils
 
-import space.taran.arknavigator.utils.provideIconImage
+import space.taran.arknavigator.mvp.model.dao.ResourceId
+import space.taran.arknavigator.mvp.model.dao.computeId
+import space.taran.arknavigator.utils.*
 import java.nio.file.Files
 import java.nio.file.Path
 
 enum class PredefinedIcon {
     FOLDER, FILE
 }
 
-data class Preview(val predefined: PredefinedIcon? = null, val image: Path? = null) {
-    companion object {
-        fun provide(path: Path): Preview {
-            if (Files.isDirectory(path)) {
-                return Preview(predefined = PredefinedIcon.FOLDER)
-            }
-
-            val image = provideIconImage(path)
-                ?: return Preview(predefined = PredefinedIcon.FILE)
+data class Preview(
+    val predefinedIcon: PredefinedIcon? = null,
+    val previewPath: Path? = null,
+    val fileType: FileType? = FileType.UNDEFINED,
```

Comment:
1. `fileType` should be named as `resourceType` since we might support non-single-file resources in future (like bookmarks #95).
2. Type of a resource should be taken from resources index as well as `extraInfo`, I'll push a commit with small modification and stubs to fill with concrete implementation.